tests/unit/wingsail/common/test_lut.py

```diff
@@ -0,0 +1,53 @@
+# import pytest
+
+import numpy as np
+from scipy import interpolate
+
+from controller.common.lut import LUT
+
+
+class TestLUT:
+    # Intialize lookup table
+    look_up_table = [[50000, 5.75], [100000, 6.75], [200000, 7], [500000, 9.75], [1000000, 10]]
+
+    def test_LUT_constructor(self):
+        # set up
+        testLUT = LUT(self.look_up_table)
+
+        # test that LUT return a known value
+        assert testLUT(50000) == 5.75
+
+    def test_unknown_interpolation_exception(self):
+        try:
+            testLUT = LUT(self.look_up_table, "gabagool")
+            assert False  # failure: constructor accepted deli meat interpolation method
+
+        except ValueError:
+            assert True
+
+        except:
+            assert False  # failure: constructor threw wrong exception
+
+    def test_linear_interpolation(self):
+        # set up
+        testLUT = LUT(self.look_up_table)
+        table = np.array(self.look_up_table)
+        test_values = list(range(50000, 1100000, 10000))
+
+        # Test that linear interpolation does not extrapolate
+        assert testLUT(10000) == 5.75
+        assert testLUT(2000000) == 10
```

Comment:
This can be a separate test. Be sure to parameterize the test with pytest parameterization.

Reply:
created seperate test:

```

    def test_linear_extrapolation(self):
        testLUT = LUT(self.look_up_table)
        # Test that linear interpolation does not extrapolate
        assert math.isclose(testLUT(10000), 5.75)
        assert math.isclose(testLUT(2000000), 10)
```